### PR TITLE
N+1 solution and specs

### PR DESCRIPTION
**Pull request by SergSkol**
- Make sure that the N+1 problem is solved when fetching all posts and their comments for a user.

Before the fix:
image

After the fix:
image

### CODE REVIEW

File: spec/views/post_index_spec.rb [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+    it 'I can see how many likes a post has' do
+      visit user_posts_path(@user)
+      expect(page).to have_content('Likes: 0')
+    end
+
+    it 'When I click on a post, it redirects me to that post`s show page' do
+      visit user_posts_path(@user)
+      click_link('Test post title')
+      expect(page).to have_current_path(user_post_path(@user, @post))
```
Comment (TedLivist) [inline spec/views/post_index_spec.rb:65]: - [ ] This test is failing. Kindly fix that.

  yrurr

**Review by TedLivist — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ ♻️

Hi @SergSkol and @kidde60,

Good job so far!
✔️ Scenarios are well covered
However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/views/post_index_spec.rb (hunk 1 above)
Comment (SergSkol) [reply spec/views/post_index_spec.rb:65]: Thank you, but there is no any error on my side. Anyway, I've made changes in "click" case, could you please check?

**Review by TedLivist — APPROVED** [review]
## STATUS: APPROVED ✔️ ✔️

Hi @SergSkol and @kidde60,

✔️ All tests are passing

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**